app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.alerts.impl;
+
+import com.arpnetworking.metrics.portal.alerts.AlertRepository;
+import com.arpnetworking.play.configuration.ConfigurationHelper;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.ExpressionList;
+import com.avaje.ebean.PagedList;
+import com.avaje.ebean.Query;
+import com.avaje.ebean.Transaction;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import models.ebean.NagiosExtension;
+import models.internal.Alert;
+import models.internal.AlertQuery;
+import models.internal.QueryResult;
+import models.internal.impl.DefaultAlert;
+import models.internal.impl.DefaultAlertQuery;
+import models.internal.impl.DefaultQuantity;
+import models.internal.impl.DefaultQueryResult;
+import org.joda.time.Period;
+import play.Configuration;
+import play.Environment;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of <code>AlertRepository</code> using SQL database.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+public class DatabaseAlertRepository implements AlertRepository {
+
+    /**
+     * Public constructor.
+     *
+     * @param environment Play's <code>Environment</code> instance.
+     * @param config      Play's <code>Configuration</code> instance.
+     * @throws Exception If the configuration is invalid.
+     */
+    @Inject
+    public DatabaseAlertRepository(final Environment environment, final Configuration config) throws Exception {
+        this(
```

Comment:
For similar reasons as in the host repository. That is, if we ever use a special kind of query like the full text search capability. I dont think we need the full search abilities on alerts but building the query generator this way at least allows the flexibility for that